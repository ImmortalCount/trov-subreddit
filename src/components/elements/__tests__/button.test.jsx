import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Button from '../button'

describe('Button Component', () => {
    const defaultProps = {
        size: 'large',
        onClick: jest.fn()
    }

    const wrapper = shallow(
        <Button {...defaultProps} />
    )

    it('should render Button component', () => {
        wrapper.simulate('click')
        expect(defaultProps.onClick).toBeCalled()
    })

    it('Button matches snapshot', () => {
        const modalRender = renderer.create(
            <Button {...defaultProps} />
        ).toJSON()
        expect(modalRender).toMatchSnapshot()
    })
})
