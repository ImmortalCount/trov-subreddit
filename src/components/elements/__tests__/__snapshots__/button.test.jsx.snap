// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button matches snapshot 1`] = `
<button
  className="sc-bdVaJa Moiob"
  color="green"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  size="large"
>
  submit
</button>
`;
